tensorboard/plugins/graph/graphs_plugin.py

```diff
@@ -170,10 +174,14 @@ def run_metadata_impl(self, run, tag):
     try:
       run_metadata = self._multiplexer.RunMetadata(run, tag)
     except ValueError:
+      # TODO(stephanwlee): Should include whether FE is fetching for v1 or v2 RunMetadata
+      # so we can remove this try/except.
       tensor_events = self._multiplexer.Tensors(run, tag)
-      # Take the first event if there are multiple events written from different
-      # steps.
-      run_metadata = RunMetadata.FromString(tensor_events[0].tensor_proto.string_val[0])
+      if len(tensor_events) > 0:
```

Comment:
This needs an else branch that also sets run_metadata otherwise it will fail at the `if run_metadata is None` because the variable won't be defined.  Either that or you can check `if not tensor_events: return None`.